snuba/utils/streams/kafka.py

```diff
@@ -658,36 +658,30 @@ def __init__(
         self.__commit_log_topic = commit_log_topic
         self.__group_id = configuration["group.id"]
 
-    def poll(self, timeout: Optional[float] = None) -> Optional[Message[TPayload]]:
-        self.__producer.poll(0.0)
-        return super().poll(timeout)
-
     def __commit_message_delivery_callback(
-        self, error: Optional[KafkaError], message: ConfluentMessage
+        self, commit: Commit, future: Future[Commit]
     ) -> None:
-        if error is not None:
-            raise Exception(error.str())
+        try:
+            message = future.result()
+        except Exception:
+            logger.warning("Failed to produce commit record: %r", commit, exc_info=True)
```

Comment:
It is still a pretty serious issue, I think this should be logged as error.

Reply:
Nah, don't worry not very strong opinion either. As long as we do not ignore these errors it is fine.